h2o-core/src/main/java/water/server/Credentials.java

```diff
@@ -0,0 +1,103 @@
+package water.server;
+
+import org.apache.commons.codec.binary.Base64;
+import water.network.SecurityUtils;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+
+/**
+ * Representation of the User-Password pair
+ */
+public class Credentials {
+  private static final int GEN_PASSWORD_LENGTH = 16;
+
+  private final String _user;
+  private final String _password;
+
+  private Credentials(String _user, String _password) {
+    this._user = _user;
+    this._password = _password;
+  }
+
+  public String toBasicAuth() {
+    return "Basic " + base64EncodeToString(_user + ":" + _password);
+  }
+
+  public String toHashFileEntry() {
+    return _user + ": " + credentialMD5digest(_password) + "\n";
+  }
+
+  public String toDebugString() {
+    return "Credentials[_user='" + _user + "', _password='" + _password + "']";
+  }
+
+  public static Credentials make(String user, String password) {
+    return new Credentials(user, password);
+  }
+
+  public static Credentials make(String user) {
+    return make(user, SecurityUtils.passwordGenerator(GEN_PASSWORD_LENGTH));
+  }
+
+  /**
+   * This replaces Jetty's B64Code.encode().
+   */
+  private static String base64EncodeToString(String s) {
+    final byte[] bytes = s.getBytes(StandardCharsets.ISO_8859_1);
+    return Base64.encodeBase64String(bytes);
+  }
+
+  // following part is copied out of Jetty's class org.eclipse.jetty.util.security.Credential$MD5, because we cannot depend on the library
+
+  private static final String __TYPE = "MD5:";
+  private static final Object __md5Lock = new Object();
+
+  private static MessageDigest __md;
+
+  /**
+   * This replaces Jetty's Credential.MD5.digest().
+   */
+  private static String credentialMD5digest(String password) {
```

Comment:
Doesn't seem to be used.

Reply:
it is used in method `Credentials.toHashFileEntry()` which seems to be called when working with `h2odriver.jar` parameters `-hash_login` and `-login_conf`. Unless we remove these, I think we need this method. I am also not sure if we can afford to replace this method with something simpler (like `hashCode()`) because this might break some use-cases of mentioned parameters.

Reply:
yep, didn't notice that! sorry for the false alarm